docs/src/apalache/assignments.md

```diff
@@ -0,0 +1,233 @@
+```tla
+VARIABLE x
+...
+
+Next == x' = x + 1
+```
+The state transition described by `Next` is fairly obvious; if `x` has the value of `4` in the current state, it will have the value of `5` in any successor state.
+This brings us to the first natural requirement by Apalache: *the transition operator must be an action.*
+
+### Successor State Encodings
+Unfortunately, the notion of an action is too broad to be a sufficient requirement for the transition operator.
+Consider this slight modification of the above example:
+```tla
+VARIABLE x, y (* new variable *)
+...
+
+Next == x' = x + 1
+```
+Just as in the first example, the expression `x' = x + 1` is, by definition, an action. 
+However, since the second example defines a state variable `y`, this action is 
+no longer a sufficient description of a relation between a current state and a successor state; it does not determine a successor value `y'`.
+This brings us to the second requirement: *the transition operator must allow Apalache to directly encode the relation between two successive states.*
+This captures two sub-requirements: firstly, we disallow transition operators which fail to specify the value of one or more variables in the successor states, like the one in the example above. Secondly, we also disallow transition operators where the value of a successor state variable is determined only by implicit equations. Consider the following two cases:
+```tla
+VARIABLE y
+...
+
+A == y' = 1
+B == y' * y' - 2 * y' + 1 = 0 
+```
+Using some basic math, we see that action `B` can be equivalently written as `(y' - 1)*(y' - 1) = 0`, so it describes the exact same successor state, in which the new value of `y` is `1`. 
+What makes it different from action `A` is the fact that this is far from immediately obvious from the syntax. 
+The fact that there happened to be a closed-form solution for which gave us an integer value for `y'`, is a lucky coincidence, as `B` could have been, for example, `y' * y' + 1 = 0`, with no real roots. 
+To avoid cases like this, we require that transition operators explicitly declare the values of state variables in successor states.
+
+We call syntactic forms, which explicitly represent successor state values, _assignment candidates_. An assignment candidate for `x` is a TLA+ expression that has one of the following forms:
+- `x' = e`, 
+- `x' \in S`,
+- `UNCHANGED x`, or
+- `x' := e` (note that `:=` is the operator defined in `Apalache.tla`)
+
+So to reformulate the second requirement: the transition operator must contain at least one assignment candidate for each variable declared in the specification.
+
+### Control Flow: Minimal and Compound Actions
+When writing non-trivial specifications, authors often end up with something similar to the following:
+```tla
+EventA == ...
+EventB == ...
+...
+
+Next == \/ EventA
+        \/ EventB
+```
+Specifically, `EventA` and `EventB` often represent mutually exclusive possibilities of execution.
```

Comment:
This kind of suggests that we do not have non-determinism, which is not true.

Reply:
I disagree, the "often" is there to suggest the exact opposite, that there are cases where this isn't true.